applications/stream-applications-core/stream-applications-security-common/src/main/java/org/springframework/cloud/stream/app/security/common/AppStarterWebSecurityAutoConfiguration.java

```diff
@@ -47,29 +59,71 @@
 @EnableWebSecurity
 public class AppStarterWebSecurityAutoConfiguration {
 
+	@Autowired
+	private WebEndpointProperties webEndpointProperties;
+
+	/*
+	 * Spring security will discover this provider. It grants the ADMIN ROLE to the admin
+	 * user, if configured, otherwise it passes through the current authentication.
+	 */
+	@Bean
+	AuthenticationProvider adminAuthenticationProvider(
+			AppStarterWebSecurityAutoConfigurationProperties securityProperties) {
+		return new AuthenticationProvider() {
+			@Override
+			public Authentication authenticate(Authentication authentication) throws AuthenticationException {
+				if (securityProperties.isEnabled() &&
+						StringUtils.hasText(securityProperties.getAdminUser()) &&
+						securityProperties.getAdminUser().equals(authentication.getName()) &&
+						StringUtils.hasText(securityProperties.getAdminPassword()) &&
+						securityProperties.getAdminPassword().equals(authentication.getCredentials().toString())) {
+
+					return new UsernamePasswordAuthenticationToken(securityProperties.getAdminUser(),
+							securityProperties.getAdminPassword(), Collections.singletonList(
+									new SimpleGrantedAuthority("ROLE_ADMIN")));
+				}
+				return authentication;
+			}
+
+			@Override
+			public boolean supports(Class<?> aClass) {
+				return UsernamePasswordAuthenticationToken.class.isAssignableFrom(aClass);
+			}
+		};
+	}
+
 	@Bean
 	WebSecurityConfigurerAdapter appStarterWebSecurityConfigurerAdapter(
 			AppStarterWebSecurityAutoConfigurationProperties securityProperties) {
 
 		return new WebSecurityConfigurerAdapter() {
-
 			@Override
 			protected void configure(HttpSecurity http) throws Exception {
+				AntPathRequestMatcher postToActuator = new AntPathRequestMatcher(
```

Comment:
One caveat of building up the path like this is that it does not handle the case where the servlet container is configured w/ a context path other than `/` (eg. '/myapp`). The matcher that EndpointRequest [ultimately leverages](https://github.com/spring-projects/spring-boot/blob/4494d3a805fa686b7dc6c56f22d5a2dd1ad978c8/spring-boot-project/spring-boot-actuator-autoconfigure/src/main/java/org/springframework/boot/actuate/autoconfigure/security/servlet/SecurityRequestMatchersManagementContextConfiguration.java#L53) does respect the path. 

If we care about that we can do something like I did in my prototype [here](https://github.com/onobc/playground/blob/main/actuator-security/src/main/java/com/example/actuator/security/WebSecurityConfig.java#L48). However, me and @dturanski discussed this and it is probably an edge case that we would not need to worry about.